### PR TITLE
Purity annotations for Math types

### PR DESCRIPTION
**Pull request by Lasser01**
### Purpose of this PR

* This PR aims to annotate all of the by-value methods in the math library.
* Closes issue #911 
* This affects the math library.

### CODE REVIEW

**Comment by Perksey** [comment]
Thanks for championing this! Give it a few days and we should have #912 merged. :)

**Comment by varon** [comment]
#912 is merged. You're free to go ahead!

**Comment by varon** [comment]
Great work so far. Looking forward to getting these in!

**Comment by Lasser01** [comment]
Thanks, I'm hoping I'll have the time to finish it this weekend.

**Comment by Perksey** [comment]
@varon Looks like this is ready for review.

**Comment by Lasser01** [comment]
Oh sorry forgot to comment, that it is ready for review

File: src/OpenToolkit.Mathematics/Matrix/Matrix3x4.cs [hunk 1]
```diff
@@ -985,7 +1011,8 @@ public override bool Equals(object obj)
         /// </summary>
         /// <param name="other">An matrix to compare with this matrix.</param>
         /// <returns>true if the current matrix is equal to the matrix parameter; otherwise, false.</returns>
-        public bool Equals(Matrix3x4 other)
+       [Pure]
+                public bool Equals(Matrix3x4 other)
```
Comment (jvbsl) [inline src/OpenToolkit.Mathematics/Matrix/Matrix3x4.cs]: wrong indentation

Comment (Lasser01) [reply src/OpenToolkit.Mathematics/Matrix/Matrix3x4.cs]: Thanks for the look through, this should now be fixed ^^

**Comment by Perksey** [comment]
Approved, just fix merge conflicts. Some methods you've added the [Pure] attribute to may have been removed in another PR.

**Comment by Lasser01** [comment]
Gonna fix that now, I'm also going to add [Pure] to the new methods.

**Comment by Lasser01** [comment]
@Perksey Merge conflicts and the new methods should be covered and fixed.

**Comment by varon** [comment]
Thanks for the excellent work @Lasser01 !

**Pull request merged**